src/game/components/Position.hpp

```diff
@@ -14,9 +14,9 @@
 namespace game
 {
     struct Position : public ecs::Component, public raylib::core::Vector3 {
-        Position(float px = 0, float py = 0, float pz = 0) : Vector3(px, py, pz) {}
-        Position(Vector3 vector) : Vector3(vector) {}
-        Position(Vector2 vector) : Vector3(vector) {}
+        Position(raylib::core::Vector2 &pposition) : raylib::core::Vector3(pposition) {}
```

Comment:
Don't pass vectors by reference, there are just two (or three) floats, so passing them in what is essentially a pointer adds needless overhead.